src/cpp/rtps/transport/shared_mem/SharedMemGlobal.hpp

```diff
@@ -0,0 +1,600 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _FASTDDS_SHAREDMEM_GLOBAL_H_
+#define _FASTDDS_SHAREDMEM_GLOBAL_H_
+
+#include <vector>
+#include <mutex>
+#include <memory>
+
+#include <rtps/transport/shared_mem/SharedMemSegment.hpp>
+#include <rtps/transport/shared_mem/MultiProducerConsumerRingBuffer.hpp>
+#include <rtps/transport/shared_mem/SharedMemLog.hpp>
+
+namespace eprosima {
+namespace fastdds {
+namespace rtps {
+
+/**
+ * This class defines the global resources for shared-memory communication.
+ * Mainly the shared-memory ports and its operations.
+ */
+class SharedMemGlobal
+{
+public:
+
+    SharedMemGlobal(
+            const std::string& domain_name)
+        : domain_name_(domain_name)
+    {
+
+    }
+
+    ~SharedMemGlobal()
+    {
+
+    }
+
+    /**
+     * Identifies a data buffer given its segment_id (shared-memory segment global_name)
+     * and offset inside the segment
+     */
+    struct BufferDescriptor
+    {
+        SharedMemSegment::Id source_segment_id;
+        SharedMemSegment::offset buffer_node_offset;
+    };
+
+    typedef MultiProducerConsumerRingBuffer<BufferDescriptor>::Listener Listener;
+    typedef MultiProducerConsumerRingBuffer<BufferDescriptor>::Cell PortCell;
+
+    struct PortNode
+    {
+        UUID<8> uuid;
+        uint32_t port_id;
+        SharedMemSegment::condition_variable empty_cv;
+        SharedMemSegment::mutex empty_cv_mutex;
+        SharedMemSegment::offset buffer;
+        SharedMemSegment::offset buffer_node;
+        std::atomic<uint32_t> ref_counter;
+        uint32_t waiting_count;
+        uint32_t check_awaken_count;
+        uint32_t check_id;
+        bool is_port_ok;
+        bool is_opened_read_exclusive;
+        bool is_opened_for_reading;
+    };
+
+    /**
+     * A shared-memory port is a communication channel where data can be written / read.
+     * A port has a port_id and a global name derived from the port_id and the domain.
+     * System processes can open a port by knowing its name.
+     */
+    class Port
```

Comment:
As a general comment on the class: Methods with doxygen documentation should have the input parameters and output values documented as well